app/models/vouchers/percentage_rate.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: false
+
+module Vouchers
+  class PercentageRate < Voucher
+    validates :amount,
+              presence: true,
+              numericality: { greater_than: 0, less_than_or_equal_to: 100 }
+
+    def display_value
+      ActionController::Base.helpers.number_to_percentage(amount, precision: 2)
+    end
+
+    # We limit adjustment to the maximum amount needed to cover the order, ie if the voucher
+    # covers more than the order.total we only need to create an adjustment covering the order.total
+    def compute_amount(order)
+      percentage = amount / 100
+      -percentage * order.pre_discount_total
+    end
```

Comment:
Was the comment meant for the clamping of the flat rate voucher?

We limit the rate to a maximum of 100% in the percentage case which can't exceed the total by definition.

Reply:
yeah it was, I'll clean that up